components/common-go/namegen/workspaceid.go

```diff
@@ -23,14 +26,26 @@ func GenerateWorkspaceID() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	s3, err := chooseRandomly(characters, 8)
+	s3, err := chooseRandomly(characters, 11)
 	if err != nil {
 		return "", err
 	}
 
 	return strings.Join([]string{s1, s2, s3}, "-"), nil
 }
 
+var (
+	InvalidWorkspaceID = errors.New("workspace id does not match required format")
+)
+
+func ValidateWorkspaceID(id string) error {
+	if !WorkspaceIDPattern.MatchString(id) {
+		return fmt.Errorf("id '%s' does not match workspace ID regex '%s': %w", id, WorkspaceIDPattern.String(), InvalidWorkspaceID)
```

Comment:
https://github.com/gitpod-io/gitpod/pull/10491#discussion_r892859657
```suggestion
		return xerrors.Errorf("id '%s' does not match workspace ID regex '%s': %w", id, WorkspaceIDPattern.String(), InvalidWorkspaceID)
```